x-pack/plugins/ml/common/types/anomaly_detection_jobs/summary_job.ts

```diff
@@ -14,32 +14,140 @@ import type { MlJobBlocked } from './job';
 export type { Datafeed } from './datafeed';
 export type { DatafeedStats } from './datafeed_stats';
 
+/**
+ * A summary of an anomaly detection job.
+ */
 export interface MlSummaryJob {
+  /**
+   * The identifier for the anomaly detection job.
+   */
   id: string;
+
+  /**
+   * A description of the job.
+   */
   description: string;
+
+  /**
+   * A list of job groups. A job can belong to no groups, one or many.
+   */
   groups: string[];
+
+  /**
+   * The number of input documents that have been processed by the anomaly detection job.
+   * This value includes documents with missing fields, since they are nonetheless analyzed.
+   */
   processed_record_count?: number;
+
+  /**
+   * The status of the mathematical models, which can take the values ok, soft_limit or hard_limit.
+   */
   memory_status?: string;
+
+  /**
+   * The status of the job.
+   */
   jobState: string;
+
+  /**
+   * An array of index names used by the datafeed. Wildcards are supported.
+   */
   datafeedIndices: string[];
+
+  /**
+   * Flag indicating whether a datafeed exists for the job.
+   */
   hasDatafeed: boolean;
+
+  /**
+   * The identifier for the datafeed.
+   */
   datafeedId: string;
+
+  /**
+   * The status of the datafeed.
+   */
   datafeedState: string;
+
+  /**
+   * The timestamp of the latest chronologically input document.
+   */
   latestTimestampMs?: number;
+
+  /**
+   * The timestamp of the earliest chronologically input document.
+   */
   earliestTimestampMs?: number;
+
+  /**
+   * The latest of the timestamp of the latest chronologically input document or the latest bucket that was processed.
+   */
   latestResultsTimestampMs?: number;
+
+  /**
+   * Used in older implementations of the job config, where the datafeed was placed inside the job for convenience.
```

Comment:
it might be worth mentioned the condition for `fullJob` being populated.
`fullJob` is populated if the job's id has been passed to the `/api/ml/jobs/jobs_summary` endpoint

Reply:
Done in https://github.com/elastic/kibana/pull/123813/commits/b9139441cbf8a47b5de9a71fa98061fb49415a38